### PR TITLE
Add documentation for Partitions initial offsets

### PR DESCRIPTION
**Pull request by k-jamroz**
Documentation for https://github.com/hazelcast/hazelcast/pull/21546

### CODE REVIEW

File: docs/modules/integrate/pages/kafka-connector.adoc [hunk 1]
```diff
@@ -75,6 +75,12 @@ committing using `enable.auto.commit` and configuring
 link:https://kafka.apache.org/22/documentation.html[Kafka documentation]
 for the descriptions of these properties.
 
+You can also explicitly specify exact initial offsets for the Kafka source using `TopicsConfig` parameter.
+Note that initial offsets provided in `topicConfig` will always have a priority over offsets stored in Kafka or associated with given consumer group.
+Those offsets are used only when the job is started for the first time after submission.
```
Comment (k-jamroz) [inline docs/modules/integrate/pages/kafka-connector.adoc:80]: technically this is until the first snapshot is taken, but IMO it is too much detail

File: docs/modules/integrate/pages/kafka-connector.adoc [hunk 2]
```diff
@@ -75,6 +75,12 @@ committing using `enable.auto.commit` and configuring
 link:https://kafka.apache.org/22/documentation.html[Kafka documentation]
 for the descriptions of these properties.
 
+You can also explicitly specify exact initial offsets for the Kafka source using https://docs.hazelcast.org/docs/{full-version}/javadoc/com/hazelcast/jet/kafka/KafkaSources.html#kafka(java.util.Properties,com.hazelcast.function.FunctionEx,com.hazelcast.jet.kafka.TopicsConfig)[`TopicsConfig` parameter^].
+Note that initial offsets provided in `topicConfig` will always have a priority over offsets stored in Kafka or associated with given consumer group.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connector.adoc:79]:
```suggestion
Note that initial offsets provided in `topicConfig` will always have priority over offsets stored in Kafka or associated with a given consumer group.
```

File: docs/modules/integrate/pages/kafka-connector.adoc [hunk 3]
```diff
@@ -75,6 +75,12 @@ committing using `enable.auto.commit` and configuring
...
+Those offsets are used only when the job is started for the first time after submission.
+Afterwards the regular fault tolerance mechanism described above is used.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connector.adoc:81]:
```suggestion
Afterwards, the regular fault tolerance mechanism described above is used. 
```

File: docs/modules/integrate/pages/kafka-connector.adoc (hunk 1 above)
Comment (rebekah-lawrence) [reply docs/modules/integrate/pages/kafka-connector.adoc:80]: I agree.

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]
Teeny amendments.
Many thanks for this @k-jamroz

**Review by rebekah-lawrence — APPROVED** [review]
LGTM

**Review by TomaszGaweda — APPROVED** [review]

**Pull request merged**